pkg/prom/api.go

```diff
@@ -0,0 +1,149 @@
+package prom
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+)
+
+const failureExitCode = -1
+
+func GetUsedExprInRules(url *url.URL) (expressions []string) {
+	promRules := GetRules(url)
+
+	for _, group := range promRules.Groups {
+		for _, r := range group.Rules {
+			switch v := r.(type) {
+			case v1.RecordingRule:
+				expressions = append(expressions, v.Query)
+			case v1.AlertingRule:
+				expressions = append(expressions, v.Query)
+			default:
+				fmt.Fprintln(os.Stderr, "error when parsing rules found rule which is not an AlertingRule nor a RecordingRule")
+				os.Exit(1)
+			}
+		}
+	}
+
+	return expressions
+}
+
+func GetRules(url *url.URL) v1.RulesResult {
+	if url.Scheme == "" {
+		url.Scheme = "http"
+	}
+	config := api.Config{
+		Address: url.String(),
+	}
+
+	// Create new client.
+	c, err := api.NewClient(config)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "error creating API client:", err)
+		return v1.RulesResult{}
+	}
+
+	// Run query against client.
+	api := v1.NewAPI(c)
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	rules, err := api.Rules(ctx)
+
+	cancel()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "error when preforming get to Rules endpoint:", err)
+		return v1.RulesResult{}
+	}
+
+	return rules
+}
+
+func SeriesPerMetric(url *url.URL, matcher string, start, end string) int {
+	if url.Scheme == "" {
+		url.Scheme = "http"
+	}
+	config := api.Config{
+		Address: url.String(),
+	}
+
+	// Create new client.
+	c, err := api.NewClient(config)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "error creating API client:", err)
+		return failureExitCode
+	}
+
+	stime, etime, err := parseStartTimeAndEndTime(start, end)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		return failureExitCode
+	}
+
+	// Run query against client.
+	api := v1.NewAPI(c)
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	val, _, err := api.Series(ctx, []string{matcher}, stime, etime) // Ignoring warnings for now.
+
+	cancel()
+	if err != nil {
+		return handleAPIError(err)
+	}
+
+	return len(val)
+}
+
+func parseStartTimeAndEndTime(start, end string) (time.Time, time.Time, error) {
+	var (
+		minTime = time.Now().Add(-9999 * time.Hour)
+		maxTime = time.Now().Add(9999 * time.Hour)
```

Comment:
maybe default to the last 2 hours since this is roughly how long active series are maintained in the TSDB head?